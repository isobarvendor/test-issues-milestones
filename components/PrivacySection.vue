<template>
<div class="privacy-component">
    <div class="privacy-section">
        <div class="primary-section">  
            <div class="header">Privacy Policy
                <p> last updated: 30 July 2020</p>
            </div>
            <div class="desc">
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
                    Pellentesque scelerisque laoreet sem a tempor. 
                    Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus, 
                    massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero. 
                    Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies 
                    egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus. 
                    Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl.
                </p>
            </div>
        </div>
        <div class="logo">
            <img src="/img/icons/coke.png"/>
        </div>
    </div>
    <div class="secondary-section">
        <div class="secondary-component">
            <div class="sub-header">Principles</div>
            <div class="desc">
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
                    Pellentesque scelerisque laoreet sem a tempor. 
                    Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus, 
                    massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero. 
                    Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies 
                    egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus. 
                    Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl. <a href="mailto:coke@sg.com"> coke@sg.com</a>
                </p>
            </div>
        </div>
         <v-expansion-panels
        v-model="panel"
        :disabled="disabled"
        multiple
        class="accordions"
        dark
      >
        <v-expansion-panel>
          <v-expansion-panel-header> Information we collect</v-expansion-panel-header>
          <v-expansion-panel-content>
            Some content
          </v-expansion-panel-content>
        </v-expansion-panel>
  
        <v-expansion-panel>
          <v-expansion-panel-header>information we collect</v-expansion-panel-header>
          <v-expansion-panel-content>
            Some content
          </v-expansion-panel-content>
        </v-expansion-panel>
  
        <v-expansion-panel>
          <v-expansion-panel-header> Information we collect</v-expansion-panel-header>
          <v-expansion-panel-content>
            Some content
          </v-expansion-panel-content>
        </v-expansion-panel>
      </v-expansion-panels>
    </div>
</div>
</template>

<script>
export default {
  name: "PrivacySection",
  components: {
  },
  data() {
    return {
      currentPage: 1
    };
  },
  props: {
    data: null,
  },
  computed:{
  },
  methods:{
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.privacy-component{
    .privacy-section{
        display: flex;

        .primary-section{
            display: flex;
            padding-bottom: 50px;
            border-bottom: 1px solid #585858;
            .header{
                width: 40%;
                p{
                    font-size: 16px;
                    margin-top: 35px;
                }
            }
            .desc{
                width: 50%;
                p{
                    line-height: 24px;
                }
            }
        }
         .logo{
            align-self: flex-end;
            margin-left: 35px;
        }
          
    }

    .secondary-section{
        .secondary-component{
            margin-top: 40px;
            display:flex;
            .sub-header{
                font-size: 25px;
                width: 26%;
            } 
            .desc{
                width: 50%;
                margin-left: 50px;
                p{
                    line-height: 24px;
                }
            } 
        }
        .accordions{
            margin: 40px 0px;
            width: 80%;
            .v-expansion-panel-header__icon{
                margin-left: auto;
            }
            .v-expansion-panel{
                background: transparent;
                color: #ffffff;
                border-bottom: 1px solid #585858;
            }
        }
        a{
            color: #FFC529;
        }
    }
    @media only screen and (max-width: 1199px) {
        .privacy-section{
            flex-direction: column-reverse;
            .primary-section{
                flex-direction: column;
                .header{
                    width:100%;
                    font-size: 18px;
                    margin-top:35px;
                    p{
                        margin-top:15px;
                        margin-left: 20px;
                    }
                }
                .desc{
                    margin-top:30px;
                    width: 100%;
                }
            }
            .logo{
                margin-left: 0px;
                align-self: center;
            }
        }
        .secondary-section{
            .secondary-component{
                .sub-header{
                    font-size: 18px;
                }
                flex-direction: column;
                .desc{
                    margin-left: 0px;
                    width: 100%;
                    margin-top: 30px;
                }
            }
            .accordions{
                width: 100%;
            }
        }
    }  
}
</style>
