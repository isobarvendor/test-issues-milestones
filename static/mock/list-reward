[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
