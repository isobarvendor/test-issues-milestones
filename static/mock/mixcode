{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
