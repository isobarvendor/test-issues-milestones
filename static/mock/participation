{"id":16,"questionId":1,"answerId":3,"participationId":17}
