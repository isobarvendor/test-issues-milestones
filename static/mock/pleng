{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}