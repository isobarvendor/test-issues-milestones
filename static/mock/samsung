{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p></p>",
      "shortDescription": "<p>ទូរស័ព្ទសាមសុង Galaxy Note20 Ultra</p>",
      "priority": 1,
      "prizeId": "ks8ns5hv",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8ns5hv/1.CAM prize-N20 Ultra@2x.png",
      "totalAmount": 20,
      "minAge": 13,
      "name": "Samsung Galaxy Note20 Ultra",
      "redemptionLink": "",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះទូរស័ព្ទសាមសុង Galaxy Note20 Ultra! សូមទាក់ទងទៅលេខ 1-800-20-2222 (មិនគិតថ្លៃសេវាទូរស័ព្ទ ២៤/៧) ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ</p>",
      "voucherCode": "CNXFUUAKAY"
    }
  }
}