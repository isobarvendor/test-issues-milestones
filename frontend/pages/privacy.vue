<template>
  <div v-if="dataStatus.status == 200" id="main" class="privacy">
    <div class="wrapper">
      <div class="container">
        <PrivacySection :data="data[0].privacy"/>
      </div>

    </div>
  </div>
  <div v-else-if="dataStatus.status >= 500">
    Status: {{dataStatus.status}}
    <br />
    {{dataStatus.message}}
  </div>
</template>

<script>
import PrivacySection from '@/components/PrivacySection'
export default {
  data(){
    return{
      dataStatus:{},
      data:null,
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],
      css: []
    };
  },
  created() {
    this.fetchData();
  },
  methods: {
    async fetchData() {
      let result = await this.$axios.get(
        "https://ayo.aircovery.com/cms-api/campaigns"
      );
      this.dataStatus = { status: result.status, message: result.statusText };
      this.data = result.data;
    },
  },
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    @media only screen and (max-width: 767px) {
      .privacy{
      }
    }
  }
</style>