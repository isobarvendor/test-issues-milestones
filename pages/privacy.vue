<template>
  <div  v-if="data" id="main" class="privacy">
    <div class="wrapper">
      <div class="container">
        <PrivacySection :data="data[0].privacy"/>
      </div>

    </div>
  </div>
</template>

<script>
import PrivacySection from '@/components/PrivacySection'
import deepClone from 'deep-clone'
export default {
  data(){
    return{
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      /*script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],*/
      css: []
    };
  },
  created() {

  },
  methods: {

  },
  computed: {
    data(){
    return deepClone(this.$store.state.CMSContent)
    },
    configData(){
      return deepClone(this.$store.state.config)
    }
},
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    @media only screen and (max-width: 767px) {
      .privacy{
      }
    }
  }
</style>
