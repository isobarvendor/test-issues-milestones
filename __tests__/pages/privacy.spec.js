import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import privacy from '@/pages/privacy.vue';

import PrivacySection from '@/components/PrivacySection'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";





let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      }
    });
  });

  const factory = () => {
    return mount(privacy, {

      localVue, store,router
    });
  };



describe('privacy', () => {


  test('renders a PrivacySection', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(PrivacySection);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
