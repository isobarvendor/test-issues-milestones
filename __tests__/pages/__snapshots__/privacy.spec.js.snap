// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`privacy renders properly 1`] = `
<div id="main" class="privacy">
  <div class="wrapper">
    <div class="container">
      <div class="privacy-component">
        <div class="privacy-section">
          <div class="primary-section">
            <div class="header"><span>Privacy Policy</span>
              <p></p>
            </div>
            <div class="desc">
              <p>We, The Coca-Cola Company and its affiliates, take our consumers’ privacy seriously. Our mission is to refresh the world and we can’t do that without your trust. This Privacy Policy explains what information we collect about you, why we collect the information, as well as how we collect and use the information so that we can deliver the best, most refreshing experience that we can.

                This Privacy Policy is intended for individuals in the United States. If you live outside of the United States and choose to use the Sites connected with this Privacy Policy, you do so at your own risk and understand that your information will be sent to and stored in the United States. By using any of our websites and mobile applications in the United States that contain a link to this Privacy Policy (collectively, “Sites”) or otherwise providing Personal Information to us, you agree to this Privacy Policy. Note that there may be other websites that reference the Coca-Cola brand but are subject to a different privacy policy.</p>
            </div>
          </div>
          <div class="logo"><img src="/img/icons/coke.png"></div>
        </div>
        <div class="secondary-section">
          <div class="secondary-component">
            <div class="sub-header">PERSONAL INFORMATION</div>
            <div class="desc">
              <p>We may use Personal Information as permitted by law, for the following business purposes:

                - to respond to your inquiries and fulfill your requests
                - to communicate with you about and to process orders, purchases, returns, services, donations, promotions, campaigns, programs, contests, sweepstakes, rewards and accounts
                - to inform you about our brands, products, promotions, events or other promotional purposes
                - to re-contact you if we have not heard from you in a while
                - to send you advertising/promotional material from any of our affiliates, and on behalf of our promotional and strategic partners
                - to improve your experience with our products and services, such as by personalizing your experience with us
                - to allow you to participate in polls, sweepstakes, promotions, contests and other promotions and to administer these activities
                - to deliver gift cards or gifts to others in accordance with your instructions
                - to permit you to participate in social sharing, including live social media feeds
                - to perform analytics, quality control, market research, and determine the effectiveness of our websites, mobile applications, promotional campaigns, and develop new products and services
              </p>
            </div>
          </div>
          <div class="accordions v-item-group theme--dark v-expansion-panels">
            <div aria-expanded="false" class="v-expansion-panel"><button type="button" class="v-expansion-panel-header">How Personal Information May Be Disclosed<transition-stub name="fade-transition">
                  <div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate material-icons theme--dark">$expand</i></div>
                </transition-stub></button>
              <transition-stub name="expand-transition">
                <!---->
              </transition-stub>
            </div>
            <div aria-expanded="false" class="v-expansion-panel"><button type="button" class="v-expansion-panel-header">THIRD PARTY SITES<transition-stub name="fade-transition">
                  <div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate material-icons theme--dark">$expand</i></div>
                </transition-stub></button>
              <transition-stub name="expand-transition">
                <!---->
              </transition-stub>
            </div>
            <div aria-expanded="false" class="v-expansion-panel"><button type="button" class="v-expansion-panel-header">How you can access or delete your Personal Information<transition-stub name="fade-transition">
                  <div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate material-icons theme--dark">$expand</i></div>
                </transition-stub></button>
              <transition-stub name="expand-transition">
                <!---->
              </transition-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
